azure-mgmt-scheduler/src/main/java/com/microsoft/azure/management/scheduler/implementation/JobImpl.java

```diff
@@ -147,4 +250,289 @@ public JobImpl withState(JobState state) {
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence()
+                    .withInterval(1));
+        }
+        this.inner().properties()
+            .recurrence()
+                .withFrequency(RecurrenceFrequency.MINUTE);
+
+        return this;
+    }
+
+    @Override
+    public JobImpl hours() {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence()
+                    .withInterval(1));
+        }
+        this.inner().properties()
+            .recurrence()
+            .withFrequency(RecurrenceFrequency.HOUR);
+
+        return this;
+    }
+
+    @Override
+    public JobImpl days() {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence()
+                    .withInterval(1));
+        }
+        this.inner().properties()
+            .recurrence()
+            .withFrequency(RecurrenceFrequency.DAY);
+
+        return this;
+    }
+
+    @Override
+    public JobImpl weeks() {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence()
+                    .withInterval(1));
+        }
+        this.inner().properties()
+            .recurrence()
+            .withFrequency(RecurrenceFrequency.WEEK);
+
+        return this;
+    }
+
+    @Override
+    public JobImpl months() {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence()
+                    .withInterval(1));
+        }
+        this.inner().properties()
+            .recurrence()
+            .withFrequency(RecurrenceFrequency.MONTH);
+
+        return this;
+    }
+
+    private void resetJobScheduleDaysList() {
+        // Only one days list can be specified, other scheduler lists need to be set to null
+        if (this.inner().properties() != null &&
+            inner().properties().recurrence() != null &&
+            inner().properties().recurrence().schedule() != null) {
+            this.inner().properties()
+                .recurrence()
+                .schedule()
+                .withWeekDays(null);
+            this.inner().properties()
+                .recurrence()
+                .schedule()
+                .withMonthDays(null);
+            this.inner().properties()
+                .recurrence()
+                .schedule()
+                .withMonthlyOccurrences(null);
+        }
+    }
+
+    @Override
+    public JobImpl onTheseDays(DayOfWeek... days) {
+        if (this.inner().properties() == null) {
```

Comment:
along lines similar to a previous comments, it seems a bunch of repeated code could be put inside a private helper method `ensureSchedule()` across the next several withers.